numpyro/contrib/nested_sampling.py

```diff
@@ -246,24 +245,37 @@ def run(self, rng_key, *args, **kwargs):
         loglik_fn = local_dict["loglik_fn"]
 
         # use NestedSampler with identity prior chain
-        prior_chain = PriorChain()
-        for name in param_names:
-            prior = UniformPrior(name + "_base", prototype_trace[name]["fn"].shape())
-            prior_chain.push(prior)
-        # XXX: the `marginalised` keyword in jaxns can be used to get expectation of some
-        # quantity over posterior samples; it can be helpful to expose it in this wrapper
-        ns = OrigNestedSampler(
-            loglik_fn,
-            prior_chain,
-            **self.constructor_kwargs,
+        def prior_model() -> PriorModelGen:
+            params = []
+            for name in param_names:
+                shape = prototype_trace[name]["fn"].shape()
+                param = yield tfpd.Uniform(low=jnp.zeros(shape), high=jnp.ones(shape), name=name + "_base")
+                params.append(param)
+            return tuple(params)
+
+        model = Model(prior_model=prior_model,
+                      log_likelihood=loglik_fn)
+
+        exact_ns = OrigNestedSampler(
+            model=model,
+            num_live_points=model.U_ndims * 25,
+            num_parallel_samplers=1,
+            max_samples=1e4,
+            uncert_improvement_patience=2,
+            **self.constructor_kwargs
```

Comment:
Could you revise this to
```
constructor_kwargs = self.constructor_kwargs.copy()
if 'num_live_points' not in self.constructor_kwargs:
    constructor_kwargs['num_live_points'] = model.U_ndims * 25
if 'max_samples' not in self.constructor_kwargs:
    constructor_kwargs['max_samples'] = 1e4
exact_ns = OrigNestedSampler(**constructor_kwargs)
```
so that users can specify those constructor kwargs

Reply:
Similar approach using:

```python
default_constructor_kwargs = dict(
    num_live_points=model.U_ndims * 25,
    num_parallel_samplers=1,
    max_samples=1e4,
    uncert_improvement_patience=2
)
default_termination_kwargs = dict(
    live_evidence_frac=1e-4
)
# Fill-in missing values with defaults. This allows user to inspect what was actually used by inspecting
# these dictionaries
list(map(lambda item: self.constructor_kwargs.setdefault(*item), default_constructor_kwargs.items()))
list(map(lambda item: self.termination_kwargs.setdefault(*item), default_termination_kwargs.items()))
```

Reply:
If you prefer a copy, let me know. In that case the user will not be able to see what was used.